pkg/operators/klusterlet/controllers/klusterletcontroller/klusterlet_controller.go

```diff
@@ -339,12 +340,17 @@ func (n *klusterletController) sync(ctx context.Context, controllerContext facto
 		}
 	}
 	// Deploy work agent
-	// scale up the work agent deployment when the hubConnectionDegraded condition is False.
-	// because the work agent deployment has a dependency of hub-kubeconfig-secret secret.
+	// do not scale up work agent pods if there is no hub-kubeconfig-secret when klusterelt CR is created, because
+	// the work agent deployment has a dependency of hub-kubeconfig-secret secret.
+	// do nod scale down work agent pods if the cluster is deleting since the work-agent needs to do some cleanup works.
```

Comment:
I would say like

work agent is scaled to 0 only when degrade is true withe reason if configmissing. It is to ensure a fast startup of work agent when the klusterlet is bootstrapped at the first time. The work agent should not be scaled to 0 in degraded condition with other reasons, because we still need work agent running even though the hub kubconfig is missing some certain permission. It can ensure work agent to clean up the resources defined in manifestworks when cluster is unjoining from the hub.

Reply:
modified.